libgalois/include/katana/ParallelSTL.h

```diff
@@ -391,6 +393,103 @@ partial_sum(InputIt first, InputIt last, OutputIt d_first) {
   }
 }
 
+template <class InputIt, class OutputIt, class UnaryOperation>
+OutputIt
+transform(
+    InputIt first, InputIt last, OutputIt d_first, UnaryOperation unary_op) {
+  // This will only work for random access iterators
+  using input_category =
+      typename std::iterator_traits<InputIt>::iterator_category;
+  using output_category =
+      typename std::iterator_traits<OutputIt>::iterator_category;
+  static_assert(
+      std::is_base_of_v<std::random_access_iterator_tag, input_category>);
+  static_assert(
+      std::is_base_of_v<std::random_access_iterator_tag, output_category>);
+
+  using diff_type = typename std::iterator_traits<InputIt>::difference_type;
+
+  diff_type length = std::distance(first, last);
+  katana::do_all(
+      katana::iterate(diff_type{0}, length),
+      [&](diff_type index) { *(d_first + index) = unary_op(*(first + index)); },
+      katana::steal());
+
+  return d_first + length;
+}
+
+template <class InputIt, class OutputIt>
+OutputIt
+copy(InputIt first, InputIt last, OutputIt d_first) {
+  // This will only work for random access iterators
+  using input_category =
+      typename std::iterator_traits<InputIt>::iterator_category;
+  using output_category =
+      typename std::iterator_traits<OutputIt>::iterator_category;
+  static_assert(
+      std::is_base_of_v<std::random_access_iterator_tag, input_category>);
+  static_assert(
+      std::is_base_of_v<std::random_access_iterator_tag, output_category>);
+
+  using diff_type = typename std::iterator_traits<InputIt>::difference_type;
+
+  diff_type length = std::distance(first, last);
+  katana::do_all(
+      katana::iterate(diff_type{0}, length),
+      [&](diff_type index) { *(d_first + index) = *(first + index); },
+      katana::steal());
+
+  return d_first + length;
+}
+
+template <class InputIt, class OutputIt, class UnaryPredicate>
+OutputIt
+copy_if(InputIt first, InputIt last, OutputIt d_first, UnaryPredicate pred) {
+  // This will only work for random access iterators
+  using input_category =
+      typename std::iterator_traits<InputIt>::iterator_category;
+  using output_category =
+      typename std::iterator_traits<OutputIt>::iterator_category;
+  static_assert(
+      std::is_base_of_v<std::random_access_iterator_tag, input_category>);
+  static_assert(
+      std::is_base_of_v<std::random_access_iterator_tag, output_category>);
+
+  using diff_type = typename std::iterator_traits<InputIt>::difference_type;
+
+  // first on_each, set ranges
+  std::vector<diff_type> prefix_sum(getActiveThreads());
+  diff_type length = std::distance(first, last);
+  katana::on_each([&](unsigned tid, unsigned total) {
+    auto [begin, end] = katana::block_range(diff_type{0}, length, tid, total);
```

Comment:
IIRC, `katana::block_range` should work on any random access iterators: `katana::block_range(first, last, tid, total)`